File: Segundo/PDOO/deepSpace/src/deepspace/NumericDamage.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author carmenc
 */
public class NumericDamage extends Damage{
    private  int nWeapons; 
    
    public NumericDamage(int weapons, int shields){
        super(shields);
        nWeapons=weapons;
    }
    
    @Override
    public Damage adjust(ArrayList<Weapon> w, ArrayList<ShieldBooster> s){
        int shields = Integer.min(s.size(), nShields);
        return new NumericDamage(Integer.min(nWeapons, w.size()), shields);
    }
    
    @Override
    public void discardWeapon(Weapon w){
        if (nWeapons>0)
                nWeapons--;
    }
    
    @Override
    public boolean hasNoEffect(){
        return nWeapons+nShields==0;
    }
    
    
    public int getNWeapons(){
        return nWeapons; 
    }
    
    @Override
    public DamageToUI getUIversion(){
        return new NumericDamageToUI(this);
    }
}

File: Segundo/Programacion Orientada a Objetos/ruby p4/NumericDamage.rb
require_relative 'Damage'
require_relative'NumericDamageToUI'

module Deepspace
    class NumericDamage < Damage
        def initialize(w,s)
            super(s)
            @nWeapons=w
        end

        def getUIversion
            NumericDamageToUI.new(self)
        end

        def copy
            copia=NumericDamage.new(@nWeapons,@nShields)
            copia
        end

        def nWeapons
            @nWeapons
        end

        def hasNoEffect
            return (@nWeapons + @nShields) == 0
        end

        def discardWeapon(w)
            if(@nWeapons > 0)
                @nWeapons -= 1
            end
        end

        def adjust(w,s)
            super
            if(w.length < @nWeapons) 
                weap=w.length
            end
            return NumericDamage.new(weap,shields)
        end
    end
end